app/src/forms/rbac/controller.js

```diff
@@ -85,7 +86,15 @@ module.exports = {
   },
   setSubmissionUserPermissions:  async (req, res, next) => {
     try {
+      const submission = await formService.read(req.query.formSubmissionId, req.currentUser);
       const response = await service.modifySubmissionUser(req.query.formSubmissionId, req.query.userId, req.body, req.currentUser);
+      if (req.body && Array.isArray(req.body.permissions) && req.query.selectedUserEmail) {
+        // Check if we are adding or removing a user from the draft invite list. empty permissions signifies that we are removing permissions from a user.
+        req.body.permissions.length ?
+          emailService.submissionAssigned(submission.form.id, response[0], req.query.selectedUserEmail, req.headers.referer).catch(() => { })
+          :
+          emailService.submissionUnassigned(submission.form.id, response[0], req.query.selectedUserEmail, req.headers.referer).catch(() => { });
```

Comment:
Doing an in-line promise style chain catch like this isn't a good idea especially when this function is already set up to be written in async/await style. It'd be better to instead let the error actually be caught by the try catch and propagate.
```js
if(req.body.permissions.length) {
  await emailService.submissionAssigned(submission.form.id, response[0], req.query.selectedUserEmail, req.headers.referer);
} else {
  await emailService.submissionUnassigned(submission.form.id, response[0], req.query.selectedUserEmail, req.headers.referer);
}
```
Remove the `await` in my example if you don't actually need to evaluate if the call actually worked or not before returning response.

Reply:
good catch, im making this an if else block instead. I will also be omitting the await call since we do not need to await them for the response.